linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/utils/SparkJobProgressUtil.scala

```diff
@@ -0,0 +1,119 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.linkis.engineplugin.spark.utils
+
+import org.apache.linkis.common.conf.CommonVars
+import org.apache.linkis.common.utils.{JsonUtils, Logging}
+import org.apache.linkis.protocol.engine.JobProgressInfo
+
+import org.apache.commons.lang3.StringUtils
+import org.apache.http.client.methods.HttpGet
+import org.apache.http.impl.client.HttpClients
+import org.apache.http.util.EntityUtils
+
+object SparkJobProgressUtil extends Logging {
+
+  val YARN_RESTFUL_URL: String =
+    CommonVars[String]("linkis.spark.once.yarn.restful.url", "").getValue
+
+  def getProgress(applicationId: String): Float = {
+    if (StringUtils.isBlank(applicationId)) return 0f
+    val sparkJobsResult = getSparkJobInfo(applicationId)
+    if (sparkJobsResult.isEmpty) return 0f
+    val tuple = sparkJobsResult
+      .filter(sparkJobResult => {
+        val status = sparkJobResult.getOrDefault("status", "").asInstanceOf[String]
+        "RUNNING".equals(status) || "SUCCEEDED".equals(status)
+      })
+      .map(result =>
+        (
+          result.get("numTasks").asInstanceOf[Integer],
+          result.get("numCompletedTasks").asInstanceOf[Integer]
+        )
+      )
+      .reduce((total, completed) => (total._1 + completed._1, total._2 + completed._2))
+    tuple._2.toFloat / tuple._1
+  }
+
+  def getSparkJobProgressInfo(applicationId: String): Array[JobProgressInfo] = {
+    val sparkJobsResult = getSparkJobInfo(applicationId)
+    if (sparkJobsResult.isEmpty) {
+      Array.empty
+    } else {
+      sparkJobsResult.map(sparkJobResult =>
+        JobProgressInfo(
+          getJobId(
+            sparkJobResult.get("jobId").asInstanceOf[Integer],
+            sparkJobResult.get("jobGroup").asInstanceOf[String]
+          ),
+          sparkJobResult.get("numTasks").asInstanceOf[Integer],
+          sparkJobResult.get("numActiveTasks").asInstanceOf[Integer],
+          sparkJobResult.get("numFailedTasks").asInstanceOf[Integer],
+          sparkJobResult.get("numCompletedTasks").asInstanceOf[Integer]
+        )
+      )
+    }
+  }
+
+  def getSparkJobInfo(applicationId: String): Array[java.util.Map[String, Object]] =
+    if (StringUtils.isBlank(applicationId)) Array.empty
+    else {
+      val getAppUrl = s"$YARN_RESTFUL_URL/ws/v1/cluster/apps/$applicationId"
```

Comment:
Is it possible to support passing in and obtaining parameters at the same time, through YARN_RESTFUL_URL.getValue(engineCreationContext.getOptions)